packages/contracts/src/AutoDriveCreditsReceiver.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {Pausable} from "@openzeppelin/contracts/utils/Pausable.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+
+contract AutoDriveCreditsReceiver is Ownable2Step, ReentrancyGuard, Pausable {
+    constructor() Ownable(msg.sender) {}
+    
+    event Deposit(bytes32 indexed intentId, uint256 depositAmount);
+    event TreasuryUpdated(address indexed oldTreasury, address indexed newTreasury);
+    event SweptToTreasury(address indexed caller, uint256 amount);
+
+    address payable public treasury;
+
+    function deposit(bytes32 intentId) public payable whenNotPaused {
+        emit Deposit(intentId, msg.value);
+    }
+
+    function setTreasury(address payable newTreasury) public onlyOwner {
+        require(newTreasury != address(0), "Invalid treasury");
```

Comment:
In `AutoDriveCreditsReceiver.md` you mention "Ensure the `treasury` is a capable receiver (EOA or contract implementing a payable fallback/receive)." but this is not implemented. I think it is a good idea.